### DISCUSSION TITLE
Ffmpeg process / memory usage

### DISCUSSION THREAD

**Discussion by djtremolo** [blakeblackshear/frigate#7608] [Ask A Question] [answered]
Frigate running on a Dell laptop with Ubuntu 22.04.3 LTS, 32GB RAM, i7 8750H

I'm wondering why the ffmpeg processes behave strangely in my Frigate setup. 

After server restart, all PIDs are in the range of 600-700. However, after running some time, the ffmpeg PID will start growing, soon reaching big numbers like 740121 or so. If I interprete this correctly, the ffmpeg process gets killed and restarted for some reason, correct?

Also, the RAM usage grows all the time, reaching 100% usage in three days, eventually it starts using the disk swapping. At that point, I typically reboot the server to give it a fresh start.

Here, my question is, is this behaviour correct or is there something wrong?

![example1](https://github.com/blakeblackshear/frigate/assets/23290782/4740f764-a58d-4aea-82f0-0dd740a0cf0d)

**Comment by NickM-27** [comment]
This is not correct, check the logs

**Reply by djtremolo** [reply]
Good point. The log was really long and I have hard time figuring out the actual problem. Please find a shortened version here: https://pastebin.com/sWDUgPNV

To briefly explain my setup:
- all cameras (Reolink 520A) are connected by the Go2Rtc running on my Frigate machine
- all Frigate cameras use the local Go2Rtc streams
- I have a secondary system (Zoneminder) running on another server. That connects to the Go2Rtc streams provided by my Frigate server.

In short: the cameras are connected only once and the stream is then multiplied by the Frigate server. I did this because I was afraid that the cameras might not handle parallel connections perfectly.

The strangest thing is that both of my setups (frigate&zoneminder) still seem to work perfectly regardless of these errors. They seem to catch all events and the live view works all the time.

**Reply by NickM-27** [reply]
What's your config? It seems that you're using http for the frigate feed which is not a typical config.

**Reply by djtremolo** [reply]
I'll get to more details tomorrow, when I get to my PC.

Anyway, I have the actual streams fetched from cameras using go2rtc and my frigate camera streams (http to 127.0.0.1) are just the local urls pointing to go2rtc streams.

Camera -> go2rtc -> frigate.

**Reply by NickM-27** [reply, marked as answer]
Yeah, I'd suggest using rtsp from go2rtc not http

**Reply by djtremolo** [reply]
Okay, I have now changed Frigate to use rtsp streams from go2rtc. I'll give it some time and get back with results.

Thank you for support!

**Reply by djtremolo** [reply]
24hrs later - still running with no issues. I'll get back after couple of days to confirm. Anyway, it's clear that using RTSP definitely has made an improvement in this case.
